core/src/display_object/movie_clip.rs

```diff
@@ -1648,12 +1648,26 @@ impl<'gc> TDisplayObject<'gc> for MovieClip<'gc> {
         point: (Twips, Twips),
     ) -> bool {
         if self.world_bounds().contains(point) {
-            for child in self.iter_execution_list() {
-                if child.hit_test_shape(context, point) {
-                    return true;
+            let mut hit_depth = 0;
+
+            for child in self.iter_render_list().rev() {
+                if child.clip_depth() > 0 {
+                    if hit_depth > 0 && child.clip_depth() >= hit_depth {
+                        if child.hit_test_shape(context, point) {
+                            return true;
+                        } else {
+                            hit_depth = 0;
+                        }
+                    }
+                } else if hit_depth == 0 && child.hit_test_shape(context, point) {
+                    hit_depth = child.depth();
```

Comment:
Somewhat minor, but: instead of iterating the render list in reverse (top-to-bottom), could you iterate through from bottom-to-top? This would let you a) immediately skip hitTesting any masked clips up to `clip_depth` if the mask itself is not hit, and b) early exit immediately when a child is hit.

Reply:
Done 👍